### DISCUSSION TITLE
Is there a limitation on the Package DAG files count in single zip?

### DISCUSSION THREAD

**Discussion by akohlislb** [apache/airflow#13067] [Q&A]
In order to manage DAG dependencies on python custom modules, I am using the Packaged DAGs feature of Airflow i.e. putting the DAG.python file in root and putting custom_modules.py in dependencies folder with empty __init__.py file. I make a zip of this and copy it to the Airflow DAG folder location. This works as expected upto a limit. If my DAG files are within 6-8 count, all the DAGs get imported successfully on the UI and function properly. When the count increases to above 8-10, the DAGs get imported to the UI, but upon clicking them, Airflow throws an error - <DAG> does not exist. When I split the DAGs into two zips, they get imported properly and can be accessed as well. Is there a upper limit to how many files can the packaged zip contain? Has anyone else faced this? 
I am running the zip packaging through a CI/CD workflow and hence, was zipping all the files in my repo together. With this limitation, I will have to zip the same dependencies again in individual zip files causing redundancy. Also, the limit is unknown. Don't know when it may break again in production.

**Comment by XD-DENG** [comment]
To the best of what I know, there is no such limit or configuration item to restrict how many DAG files you can have in a single ZIP.

May you share a bit more information, like the log or traceback?

**Reply by akohlislb** [reply]
I checked the logs. The issue is for the dependent custom modules in the dependencies folder. I have two custom modules in the folder. And in my DAG files, I import them one after another with the following syntax:
from dependencies.CommonServices import CommonServices
from dependencies.Utilities import Utilities

When I check the logs of Airflow Web, the log reads:
**[2020-12-16 17:00:03,125] {{dagbag.py:240}} ERROR - Failed to import: /usr/local/airflow/dags/packaged-dags.zip
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/dagbag.py", line 237, in process_file
    m = importlib.import_module(mod_name)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "/usr/local/airflow/dags/packaged-dags.zip/Database_Load_DAG.py", line 13, in <module>
    from dependencies.Utilites import Utilites
ModuleNotFoundError: No module named 'dependencies.Utilites'**

However above this line, I already have similar import statement which is getting imported without any error.
